react/Text/Text.js

```diff
@@ -25,35 +27,40 @@ const Text = ({
   light,
   baseline,
   ...restProps
-}) => (
-  <span
-    {...restProps}
-    className={classnames({
-      [styles.root]: true,
-      [className]: className,
-      [styles[size]]: size,
-      [styles.raw]: raw,
-      [styles.baseline]: baseline
-    })}>
-    <span
+}) => {
+  const Component = component || (bullet ? 'li' : 'span');
```

Comment:
If we add `component` to the props list with a default of `span` this can be simplified to: `bullet ? 'li' : component`

Reply:
This would change the behaviour.
The current behaviour priorises a consumer defined component over a bullet's defaut component.
This was chosen to keep things be a little less magical for the consumer, sometimes honouring a custom component and sometimes not may lead to confusion while debugging.